src/openforms/payments/contrib/ogone/plugin.py

```diff
@@ -141,7 +148,10 @@ def apply_status(self, payment, ogone_status) -> None:
         qs = SubmissionPayment.objects.filter(id=payment.id)
         qs = qs.exclude(status__in=PAYMENT_STATUS_FINAL)
         qs = qs.exclude(status=new_status)
-        res = qs.update(status=new_status)
+        if payment_id:
+            res = qs.update(status=new_status, provider_payment_id=payment_id)
+        else:
+            res = qs.update(status=new_status)
```

Comment:
```suggestion
        update_fields = {"status": new_status}
        if payment_id:
            update_fields["provider_payment_id"] = payment_id
        res = qs.update(**update_fields)
```
I'm a bit paranoid about different code paths/branches where some part needs to be kept in sync between all of them, and in general I prefer flattening the code such that "forgetting" things should be made impossible in the first place